addons/mail/static/src/new/core/partner_model.js

```diff
@@ -126,6 +126,16 @@ export class Partner {
                     return 1;
                 }
             }
+            if (thread) {
+                const isAFollower = thread.followers.some((follower) => follower.partner === a);
+                const isBFollower = thread.followers.some((follower) => follower.partner === b);
+                if (isAFollower && !isBFollower) {
+                    return -1;
+                }
+                if (!isAFollower && isBFollower) {
+                    return 1;
+                }
+            }
```

Comment:
Search is better in chatter. Could be even better to put other followers before self.